### PR TITLE
Remove deprecated code

### LINKED ISSUE #13262: Remove deprecated code in `main`
### Description

We have some code that has been deprecated for a long time, see e.g. `IntField` constructors. Let's remove this deprecated code in the `main` branch ahead of 10.0.

Note: we may have code that is only marked as deprecated in `main`. This code shouldn't be removed as users will get the deprecation warning for the first time when they upgrade to 10.0.

### PR DESCRIPTION
**Pull request by iamsanjay**
### Description

closes #13262 

Additionally I removed deprecated code from two more classes: VH_BE_DOUBLE (static member in BitUtil) and variant of BlockJoinSelector.wrap method. Both classes are tagged with @lucene.internal and I searched in Solr and elasticSearch. Let me know If any potential issues arise from its removal, I'll reinstate it.

### CODE REVIEW

**Comment by jpountz** [comment]
Thanks for looking into this. I think I'd rather keep `VH_BE_DOUBLE` for now for consistency, since we're also keeping the INT, LONG and FLOAT BE variants.

 - In addition to IntField, can you also remove deprecated constructors from other `*Field` classes such as `DoubleField`?
 - Can you also remove deprecated code in `BooleanQuery` and `ByteBuffersDataInput`?

**Comment by iamsanjay** [comment]
I looked into different Search engines whether they use it or not  -- Many of them still using the deprecated code.
1. `LongPoint.newDistanceFeatureQuery` (ElasticSearch uses it https://github.com/search?q=repo%3Aelastic%2Felasticsearch+LongPoint.newDistanceFeatureQuery&type=code and OpenSearch as well https://github.com/search?q=repo%3Aopensearch-project%2FOpenSearch%20newDistanceFeatureQuery&type=code) 
2.  BooleanQuery.TooManyClauses (Elastic Search https://github.com/search?q=repo%3Aelastic%2Felasticsearch%20TooManyClauses&type=code)
3. BooleanQuery.getMaxClauseCount and BooleanQuery.setMaxClauseCount are still in use.

Should I go ahead and remove it?

**Comment by jpountz** [comment]
@iamsanjay Yes please go ahead. All these problems should be easy to address by these various projects. Thanks for checking!

**Comment by iamsanjay** [comment]
I added MIGRATE docs. Let me know if that's not required.
There is one more class pending `SortField`. I am still looking into some of the test cases that are failing , we may have to remove those test cases along with deprecated code. We can also deal with it in separate pull request.

**Review by jpountz — APPROVED** [review]
I'm not sure how to handle the `SortField` deprecations either, but what you have here looks pretty good to me.

File: lucene/CHANGES.txt [hunk 1]
```diff
@@ -98,6 +98,9 @@ API Changes
 
 * GITHUB#13241: Remove Accountable interface on KnnVectorsReader. (Pulkit Gupta)
 
+* GITHUB#13262: Remove deprecated code IntField.IntField(name, value), VH_BE_DOUBLE (static member in BitUtil) and
```
Comment (jpountz) [inline lucene/CHANGES.txt:101]: Can you update the CHANGES entry?

**Review by jpountz — APPROVED** [review]
This is great.

**Pull request merged**